src/screen.cc

```diff
@@ -781,12 +809,32 @@ std::string CScreen::choose_string(std::vector<std::string> choices)
                 selected -= cols;
 
         }
+        /*
+         * Update the search string and our matches.
+         */
+        if (c == KEY_BACKSPACE || isprint(c))
+        {
+            if (c == KEY_BACKSPACE && search_string.length() > 0)
+                search_string.pop_back();
+            else
+                search_string += c;
+
```

Comment:
This needs some work.

* Open the prompt, by pressing `:`.
* Type "Screen[TAB]"
* The list comes up.
* Press "`BACKSPACE`"
* The completion-text shows "^G", and all the choices disappear.

Perhaps:


            if (c == KEY_BACKSPACE ) {
                   if ( search_string.length() > 0)
                         search_string.pop_back();
                   else 
                         beep();
            }
            else {
                search_string += c;
            }

Reply:
Sorry for this I had it right but than I wanted to be fancy and only use one if statement.

Reply:
No worries, I'm paranoid about completion code at the moment so I had to test it a bit more than I usually would.

It's a nice change, thank you :)